test/common/buffer/owned_impl_test.cc

```diff
@@ -39,6 +39,38 @@ class OwnedImplTest : public BufferImplementationParamTest {
 INSTANTIATE_TEST_CASE_P(OwnedImplTest, OwnedImplTest,
                         testing::ValuesIn({BufferImplementation::Old, BufferImplementation::New}));
 
+TEST_P(OwnedImplTest, CopyConstructAndAssign) {
+  std::string str("hello world");
+  Buffer::OwnedImpl buffer;
+  verifyImplementation(buffer);
+  buffer.add(str);
+  Buffer::OwnedImpl buffer2(buffer);
+  EXPECT_EQ(str, buffer2.toString());
+  buffer2.drain(str.length());
+  EXPECT_EQ(str, buffer.toString());
+
+  Buffer::OwnedImpl buffer3;
+  buffer3.add("aaa");
+  buffer3 = buffer;
+  EXPECT_EQ(str, buffer3.toString());
+  buffer3.drain(str.length());
+  EXPECT_EQ(str, buffer.toString());
+}
+
+TEST_P(OwnedImplTest, MoveConstructAndAssign) {
+  std::string str("hello world");
```

Comment:
you can factor out the block of code that tests a buffer into the test-base. Then call that helper from multiple tests.

That will reduce the amount of repeated code in this test.

Reply:
The test part are more or less different. I extracted the impl verification and population into a helper.